stake-pool/program/tests/helpers/mod.rs

```diff
@@ -0,0 +1,732 @@
+                &id(),
+            ),
+            instruction::initialize(
+                &id(),
+                &stake_pool.pubkey(),
+                owner,
+                &validator_stake_list.pubkey(),
+                pool_mint,
+                pool_token_account,
+                &spl_token::id(),
+                init_args,
+            )
+            .unwrap(),
+        ],
+        Some(&payer.pubkey()),
+    );
+    transaction.sign(
+        &[payer, stake_pool, validator_stake_list],
+        *recent_blockhash,
+    );
+    banks_client.process_transaction(transaction).await.unwrap();
+}
+
+pub async fn create_vote(
+    banks_client: &mut BanksClient,
+    payer: &Keypair,
+    recent_blockhash: &Hash,
+    vote: &Keypair,
+) {
+    let rent = banks_client.get_rent().await.unwrap();
+    let rent_voter = rent.minimum_balance(VoteState::size_of());
+
+    let mut transaction = Transaction::new_with_payer(
+        &[system_instruction::create_account(
+            &payer.pubkey(),
+            &vote.pubkey(),
+            rent_voter,
+            VoteState::size_of() as u64,
+            &solana_vote_program::id(),
+        )],
+        Some(&payer.pubkey()),
+    );
+    transaction.sign(&[&vote, payer], *recent_blockhash);
+    banks_client.process_transaction(transaction).await.unwrap();
+}
+
+pub async fn create_independent_stake_account(
+    banks_client: &mut BanksClient,
+    payer: &Keypair,
+    recent_blockhash: &Hash,
+    stake: &Keypair,
+    authorized: &stake::Authorized,
+    lockup: &stake::Lockup,
+) -> u64 {
+    let rent = banks_client.get_rent().await.unwrap();
+    let lamports =
+        rent.minimum_balance(std::mem::size_of::<stake::StakeState>()) + TEST_STAKE_AMOUNT;
+
+    let mut transaction = Transaction::new_with_payer(
+        &stake::create_account(
+            &payer.pubkey(),
+            &stake.pubkey(),
+            authorized,
+            lockup,
+            lamports,
+        ),
+        Some(&payer.pubkey()),
+    );
+    transaction.sign(&[payer, stake], *recent_blockhash);
+    banks_client.process_transaction(transaction).await.unwrap();
+
+    lamports
+}
+
+pub async fn create_blank_stake_account(
+    banks_client: &mut BanksClient,
+    payer: &Keypair,
+    recent_blockhash: &Hash,
+    stake: &Keypair,
+) -> u64 {
+    let rent = banks_client.get_rent().await.unwrap();
+    let lamports = rent.minimum_balance(std::mem::size_of::<stake::StakeState>()) + 1;
+
+    let mut transaction = Transaction::new_with_payer(
+        &[system_instruction::create_account(
+            &payer.pubkey(),
+            &stake.pubkey(),
+            lamports,
+            std::mem::size_of::<stake::StakeState>() as u64,
+            &stake::id(),
+        )],
+        Some(&payer.pubkey()),
+    );
+    transaction.sign(&[payer, stake], *recent_blockhash);
+    banks_client.process_transaction(transaction).await.unwrap();
+
+    lamports
+}
+
+pub async fn create_stake_account(
```

Comment:
If you do not mind, I'll leave it to the next PR, now we're doing a bunch of negative tests which have a lot of changes in the test module, so it would be cleaner to do this after the tests are done.

Reply:
Ok no problem